client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c

```diff
@@ -471,100 +468,106 @@ kaa_error_t kaa_tcp_channel_set_access_point(void *context, kaa_access_point_t *
-        KAA_LOG_ERROR(channel->logger, KAA_ERR_INSUFFICIENT_BUFFER, "Kaa TCP channel new access point [0x%08X], "
-                "insufficient connection data length  %zu, position %zu",
-                    channel->access_point.id,
-                    connection_data_len,
-                    (position + remaining_to_read));
-        return KAA_ERR_INSUFFICIENT_BUFFER;
+    /* Read hostname length */
+    remaining_to_read = sizeof(uint32_t);
+    if ((position + remaining_to_read) > connection_data_len) {
+      goto error_insufficient_buffer;
     }
+    channel->access_point.hostname_length = get_uint32_t(connection_data + position);
+    position += remaining_to_read;
 
+    /* Read hostname */
+    remaining_to_read = channel->access_point.hostname_length;
+    if ((position + remaining_to_read) > connection_data_len) {
+      goto error_insufficient_buffer;
+    }
+    channel->access_point.hostname = (char *) KAA_MALLOC(channel->access_point.hostname_length);
+    if (!channel->access_point.hostname) {
+        KAA_FREE(channel->access_point.public_key);
+        return KAA_ERR_NOMEM;
+    }
 
+    memcpy(channel->access_point.hostname, connection_data + position, remaining_to_read);
+    position += remaining_to_read;
 
-    remaining_to_read = 4;
+    /* Read access point's port */
+    remaining_to_read = sizeof(uint32_t);
     int access_point_socket_port = 0;
-    if ((position + remaining_to_read) <= connection_data_len) {
-        access_point_socket_port = (uint16_t) get_uint32_t(connection_data + position);
-        position += remaining_to_read;
-    } else {
-        KAA_LOG_ERROR(channel->logger, KAA_ERR_INSUFFICIENT_BUFFER, "Kaa TCP channel new access point [0x%08X], "
-                "insufficient connection data length  %zu, position %zu",
-                    channel->access_point.id,
-                    connection_data_len,
-                    (position + remaining_to_read));
-
-        return KAA_ERR_INSUFFICIENT_BUFFER;
+    if ((position + remaining_to_read) > connection_data_len) {
+      goto error_insufficient_buffer;
     }
+    access_point_socket_port = (uint16_t) get_uint32_t(connection_data + position);
+    position += remaining_to_read;
     channel->access_point.port = (uint16_t) access_point_socket_port;
 
 #ifdef KAA_LOG_LEVEL_TRACE_ENABLED
-    char ap_hostname[channel->access_point.hostname_length + 1];
-    memcpy(ap_hostname, channel->access_point.hostname, channel->access_point.hostname_length);
-    ap_hostname[channel->access_point.hostname_length] = '\0';
-
-    KAA_LOG_TRACE(channel->logger, KAA_ERR_NONE, "Kaa TCP channel new access point [0x%08X] destination %s:%d",
-            channel->access_point.id,
-            ap_hostname,
-            access_point_socket_port);
+    uint8_t *ap_hostname = (uint8_t *) KAA_MALLOC(channel->access_point.hostname_length + 1);
+    if (ap_hostname) {
+        memcpy(ap_hostname, channel->access_point.hostname, channel->access_point.hostname_length);
```

Comment:
If you really-really want to print AP name, you can use `printf`'s width specifier